packages/ozone/src/daemon/scheduled-action-processor.ts

```diff
@@ -0,0 +1,243 @@
+import { Selectable } from 'kysely'
+import { MINUTE, SECOND } from '@atproto/common'
+import {
+  assertProtectedTagAction,
+  getProtectedTags,
+} from '../api/moderation/util'
+import { Database } from '../db'
+import { ScheduledAction } from '../db/schema/scheduled-action'
+import {
+  ModEventTakedown,
+  ModTool,
+} from '../lexicon/types/tools/ozone/moderation/defs'
+import { dbLogger } from '../logger'
+import { ModerationService, ModerationServiceCreator } from '../mod-service'
+import { RepoSubject } from '../mod-service/subject'
+import { ModEventType } from '../mod-service/types'
+import { ScheduledActionServiceCreator } from '../scheduled-action/service'
+import { SettingService, SettingServiceCreator } from '../setting/service'
+
+export class ScheduledActionProcessor {
+  destroyed = false
+  processingPromise: Promise<void> = Promise.resolve()
+  timer?: NodeJS.Timeout
+
+  constructor(
+    private db: Database,
+    private serviceDid: string,
+    private settingService: SettingServiceCreator,
+    private modService: ModerationServiceCreator,
+    private scheduledActionService: ScheduledActionServiceCreator,
+  ) {}
+
+  start() {
+    this.poll()
+  }
+
+  poll() {
+    if (this.destroyed) return
+    this.processingPromise = this.findAndExecuteScheduledActions()
+      .catch((err) =>
+        dbLogger.error({ err }, 'scheduled action processing errored'),
+      )
+      .finally(() => {
+        this.timer = setTimeout(() => this.poll(), getInterval())
+      })
```

Comment:
This will set the timer again even if `ScheduledActionProcessor` has been destroyed.

Reply:
hmm... i copied this from the event-reverser worker https://github.com/bluesky-social/atproto/blob/main/packages/ozone/src/daemon/event-reverser.ts so didn't pay too much attention.
even if it does, wouldn't it just poll 1 extra time and then end it because of the `if (this.destroyed) return` check?